commons-test/src/main/java/org/infinispan/commons/test/PolarionTestNameListenerAdapter.java

```diff
@@ -0,0 +1,127 @@
+package org.infinispan.commons.test;
+
+import java.lang.reflect.Field;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+import org.testng.TestListenerAdapter;
+import org.testng.annotations.Test;
+
+/**
+ * Friendly test name
+ *
+ * @author <a href='mailto:afield[at]redhat[dot]com'>Alan Field</a>
+ * @author <a href='mailto:dlovison[at]redhat[dot]com'>Diego Lovison</a>
+ */
+public class PolarionTestNameListenerAdapter extends TestListenerAdapter {
+
+   @Override
+   public void onTestStart(ITestResult result) {
+      fixTestName(result);
+      super.onTestStart(result);
+   }
+
+   @Override
+   public void onTestSuccess(ITestResult result) {
+      super.onTestSuccess(result);
+   }
+
+   @Override
+   public void onTestFailure(ITestResult result) {
+      super.onTestFailure(result);
+   }
+
+   @Override
+   public void onTestSkipped(ITestResult result) {
+      super.onTestSkipped(result);
+   }
+
+   @Override
+   public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
+      super.onTestFailedButWithinSuccessPercentage(result);
+   }
+
+   @Override
+   public void onStart(ITestContext context) {
+      super.onStart(context);
+   }
+
+   @Override
+   public void onFinish(ITestContext context) {
+      super.onFinish(context);
+   }
+
+   private void fixTestName(ITestResult result) {
+      try {
+         Field xField = result.getClass().getDeclaredField("m_name");
```

Comment:
It's a hack, but I guess it will work... you just need to obtain the field and cache it in a static block.